### DISCUSSION TITLE
MissionControl 0.9.1

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#607] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.9.1/total">
<a href="https://discord.gg/gegfNZ5Ucz"><img alt="Discord Server" src="https://discordapp.com/api/guilds/905069757361971303/widget.png?style=shield"/></a>
</p>

This is the eighteenth official release of Mission Control.

I said it wasn't a priority. I said it was likely going to be too much work and probably wouldn't happen. Well, I guess I was wrong because Mission Control **now supports Dualshock 3** controllers!

- Pairing Dualshock 3 controllers works a little differently to regular Bluetooth controllers as it requires the use of a USB cable. 
- To pair the controller, you must be on the `Change Grip/Order` screen as usual.
- Connect the controller to the console via USB cable (you will need a USB-C OTG adapter if in handheld mode).
- Once the red player LEDs start flashing, remove the USB cable and press the `PS` button. Wait a few seconds and you should see the usual `Paired` notification to indicate the controller is now paired with the console. 
- You can reconnect to the console from now on by simply pressing the `PS` button.

This release also makes some changes to the settings in the `missioncontrol.ini` configuration file. Please refer to the [readme](https://github.com/ndeadly/MissionControl#module-configuration) for full details

- Added `dualshock3_led_mode` setting to select between Switch and PS3 style player LED patterns.
- Added `dualshock4_polling_rate` to modify the polling rate used by Dualshock 4 controllers.
- `enable_dualshock4_lightbar` -> `dualshock4_lightbar_brightness`
  `enable_dualsense_lightbar` -> `dualsense_lightbar_brightness`
  These two settings were modified from a boolean to enable/disable the LED lightbar to an integer value controlling brightness.
- `Renamed enable_dualsense_player_leds` -> `dualsense_enable_player_leds` to be consistent with the other settings names

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl#readme) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

### Changelog:
- Added support for Sony Dualshock 3 controller.
- Added support for Mocute 053 controller.
- Added .ini config option to select preferred Dualshock 3 player LED pattern.
- Added .ini config option to control the polling rate used by Dualshock 4 controllers.
- Replaced .ini config option for enabling/disabling the LED lightbar on Sony controllers with a brightness control setting.
- Renamed some of the other .ini config options.
- Wiimote dpad is now mapped to the left analog stick when in horizontal orientation to improve compatibility with games that don't allow dpad inputs for movement.

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.9.1'>MissionControl 0.9.1</a>.</em>

**Comment by marionicolas** [comment]
Hi!!!! its a incredible update!! im just testing it and wodering, should it work with non official ds3 controllers?

**Reply by ndeadly** [reply]
It might, but I don't make any guarantees. I only tested with an authentic one, and I know that many of the clones have quirks that require their own workarounds.

**Reply by marionicolas** [reply]
Im testing a generic ps3 controller, made the pairing, but when trying to connect bt, first flash fast, a second after flashes slow... but keeps flashing slow for ever.

**Reply by marionicolas** [reply]
Tell me if this is not the place, im sorry never user github for more than downloading or starring hahaha

**Reply by ndeadly** [reply]
If it flashes for a long time without switching off then it might be connected. A command needs to be sent to switch the LEDs to a solid player indicator, it's possible this fails. If it times out after a while it's probably just searching for a host though.

> Tell me if this is not the place, im sorry never user github for more than downloading or starring hahaha

Here is fine, but if you want this to be treated as more than just a conversation I suggest you open a bug report issue for it. I can't guarantee I can do anything for you, not owning the controller myself, but at least that way it has high visibility and won't get forgotten about, and may encourage input from other users too.

**Reply by ndeadly** [reply]
You can try making a log with this (it will create one in `sdmc:/mc-mitm.log`). I can only log pretty high level events with it, but it may give some clues regarding whether the controller is actually connecting or not.

[MissionControl-0.9.1-debug-logging-05bbc5b.zip](https://github.com/ndeadly/MissionControl/files/11311080/MissionControl-0.9.1-debug-logging-05bbc5b.zip)

**Reply by marionicolas** [reply]
Ok, i will try to get that log and open e Issue with... you are awesome man!

**Reply by marionicolas** [reply]
https://github.com/ndeadly/MissionControl/issues/615

Im reading further on this, i use this controller with esp32 (arduino) and it seems to work well with all the genuine ps3 protocol, so ill keep doing testings.

Thanks a lot!!!

**Reply by nunits** [reply]
I have not been able to make the ps3 controller flash red leds when connected. it is possible that it may be blocked from having sys with tesla modules
When it asks to restart, can only restart payload ?

**Reply by ndeadly** [reply]
@nunits Is it an official controller or a clone? I can't guarantee clone behaviour is going to be the same as the official controller. I can't remember whether it's MC that triggers the flashing, or just the behaviour when you disconnect the USB cable either. It's possible for other homebrew to interfere with this if they set up the same USB interface filter. As far as I know none of the existing homebrew does this though. When in doubt you could always disable anything else you have running to be sure.

> When it asks to restart, can only restart payload ?

Don't know what you mean by this.

**Reply by marionicolas** [reply]
> When it asks to restart, can only restart payload ?

If you mean when you enable Mission Control on tesla menu, yes, it is enough to restart to payload.

**Reply by nunits** [reply]
> @nunits Is it an official controller or a clone? I can't guarantee clone behaviour is going to be the same as the official controller. I can't remember whether it's MC that triggers the flashing, or just the behaviour when you disconnect the USB cable either. It's possible for other homebrew to interfere with this if they set up the same USB interface filter. As far as I know none of the existing homebrew does this though. When in doubt you could always disable anything else you have running to be sure.
> 
> > When it asks to restart, can only restart payload ?
> 
> Don't know what you mean by this.

It is an original control, I disabled some modules but I have not had an answer. The control when connecting and disconnecting it from usb does not turn on the flashing led. It only turns on if I press the PS button. The usb works to charge other devices but apparently it doesn't charge the ps3 controller.
When connected to a PC, the LEDs work.
I'll keep trying.

**Reply by ndeadly** [reply]
How are you connecting it to the console? Maybe it's an issue with the USB cable or adapter (if in handheld mode). Also, are you sure that Mission Control is running? You can install the debug build I linked a few posts back if you want. The log will tell me a) if Mission Control is running, and b) what's going on regarding USB and the DS3 controller.

**Reply by nunits** [reply]
> como lo conectas a la consola? Tal vez sea un problema con el cable USB o el adaptador (si está en modo portátil). Además, ¿estás seguro de que Mission Control se está ejecutando? Puede instalar la compilación de depuración que vinculé algunas publicaciones si lo desea. El registro me dirá a) si Mission Control se está ejecutando yb) qué está pasando con el USB y el controlador DS3.

I have tested it by connecting to the 2 usb of the dock and directly to the console with a dogleusb
Both the ps3 and the ps4 control do not charge the battery by connecting them. But if it works charging battery ps5 control.
I remember that a while ago I could charge all the controllers from the dock ports.
Maybe something is wrong with this and it makes it so that I can't bind the ps3 controller.
I've tried replacing the mission control files, but I'm getting A fatal error occurred while running Atmosphere. Title ID: 0100000000000008
I delete the folder atmosphere/contents/010000000000bd00
and you can start the console, I enter the HB APP Store to install mission control fine, I can use ps4 and ps5 control without problem.
Something I can do or it only remains to clean the entire SD.

**Reply by ndeadly** [reply]
@nunits 0100000000000008 is atmosphere's boot2. I got this error at some point when I was testing, I think it was because something didn't copy to SD correctly. I would try reinstall both atmosphere and Mission Control. The build above works, I have been using it and so have several others without issues. Without a log I can't really help your further.

**Reply by nunits** [reply]
I have made a clean copy of both atmosphere and mission control.
But I have not been able to solve the connection problem with ps3 control.
now I had no problem to replace mission control and I was able to get a log
-Try linking ps3 controller...
-Pair ps4 controller correctly
-Try again to link ps3 controller.
[mc-mitm.log](https://github.com/ndeadly/MissionControl/files/11337180/mc-mitm.log)

**Reply by ndeadly** [reply]
This log shows no evidence of USB communications at all. This suggests a problem somewhere along the USB chain. Could be your switch, dock, cable or even controller. Also, not sure if you need to enable `Pro Controller Wired Communication` in system settings. Probably not, since it's not setting up an hid connection, but I never tested with it disabled.

**Reply by ndeadly** [reply]
Also, this only works from the `Change Grip/Order` screen. You are trying to pair from there, right?

**Reply by nunits** [reply]
> Also, this only works from the `Change Grip/Order` screen. You are trying to pair from there, right?

Yes, I try from that place. But my apologies because in the end I have concluded that my nintendo switch has problems detecting usb controllers. I only receive charge. surely the dock has problems even so thank you very much, for the kindness of responding quickly and especially for your work.
